test/sbml-test-suite/STS.cpp

```diff
@@ -0,0 +1,104 @@
+//
+// Created by Ciaran on 13/12/2021.
+//
+
+#include "STS.h"
+#include <filesystem>
+
+std::string errorMsg(const std::string& varName){
+    std::ostringstream os;
+    os << "The \""<<varName<<"\" variable is not defined. Make sure you have added it to your library or "
+                             "executable using cmake's target_compile_options command";
+    return os.str();
+}
+
+STS::STS() {
+#ifndef SBMLTestSuiteRoot
+    throw std::logic_error(errorMsg("SBMLTestSuiteRoot"));
+#endif
+#ifndef SBMLTestSuiteSemantic
+    throw std::logic_error(errorMsg("SBMLTestSuiteSemantic"));
+#endif
+#ifndef SBMLTestSuiteStochastic
+    throw std::logic_error(errorMsg("SBMLTestSuiteStochastic"));
+#endif
+    if (!std::filesystem::exists(SBMLTestSuiteRoot)) {
+        throw std::invalid_argument("Path to SBMLTestSuiteRoot does not exist");
+    }
+    if (!std::filesystem::exists(SBMLTestSuiteSemantic)) {
+        throw std::invalid_argument("Path to SBMLTestSuiteSemantic does not exist");
+    }
+    if (!std::filesystem::exists(SBMLTestSuiteStochastic)) {
+        throw std::invalid_argument("Path to SBMLTestSuiteStochastic does not exist");
+    }
+
+    Logger::setLevel(Logger::LOG_WARNING);
+
+}
+
+std::string STS::constructSTSXmlFile(int caseId, int level, int version) {
+    std::ostringstream caseOs;
+    std::ostringstream fname;
+
+    caseOs << std::setfill('0') << std::setw(5) << caseId;
+    fname << caseOs.str() << "-sbml-l" << level << "v" << version << ".xml";
+
+    std::filesystem::path root = SBMLTestSuiteSemantic;
+    std::filesystem::path caseDir = root / caseOs.str();
+    std::filesystem::path xmlFile = caseDir / fname.str();
+    if (!std::filesystem::exists(xmlFile)) {
+        std::ostringstream os;
+        os << "file not found : " << xmlFile;
+        throw std::logic_error(os.str());
+    }
+    return xmlFile.string();
+}
+
+std::vector<std::string> STS::getFirstNModelsFromSTS(int N, int start) {
```

Comment:
I'm not sure how you're expecting this function to be used, but 'first N' seems highly specific.  Can you instead take just a list or a set or a range iterator as the argument, and return the tests that match the elements of that list?  Or just make it 'getModelNFromSTS' and return the single XML file, and let the user set up 'first N' or 'test x' or 'tests x-y' or 'tests x, y, and z' or 'all tests with the tag XXX', etc.?

Reply:
I just needed something quickly to get going with [RoadRunnerMapTests.cpp](https://github.com/sys-bio/roadrunner/blob/80a3bc2c92648b131aa23f5091cc7a2b3b61739e/test/cxx_api_tests/RoadRunnerMapTests.cpp#L30) which loads models in parallel. Since I wasn't testing specific models or sbml features I didn't actually care about the identity of the models being loaded, hence "firstN". [We already have the [getModelNFromSTS](https://github.com/sys-bio/roadrunner/blob/56c78ae3b037eca6dfe210b1fc97faa2060b028b/test/sbml-test-suite/STS.h#L19), I'll refactor the name. 

The interface for this can definitely be improved. In general, I would like small but nice and easy to use couple of methods that provide access to the STS for unit/integration testing. The methods you've suggested make sense to me.

Reply:
Sure; how about we file an issue and leave it as-is for now?

Reply:
I kind of accidently spent today on this. Actually I don't mind because its something I've wanted to do for a while. First draft will be checked in today -- feel free to modify as you see fit.

Reply:
Ha!  I understand that. Obviously not a problem; happy to look at the updated version.